api/jobs.py

```diff
@@ -267,3 +280,8 @@ def put(self, _id):
         result = self.app.db.jobs.update_one(job_query, {'$set': mutation})
         if result.modified_count != 1:
             self.abort(500, 'Job modification not saved')
+
+        # If the job did not succeed, check to see if job should be retried.
+        if 'state' in mutation and mutation['state'] == 'failed':
```

Comment:
Standard for all [maps](https://docs.python.org/2/library/stdtypes.html#mapping-types-dict).